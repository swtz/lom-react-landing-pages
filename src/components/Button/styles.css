.btn {
  width: 100%;
  background: darkblue;
  color: white;
  border: none;
  padding: 15px 30px;
  cursor: pointer;
}

.btn:disabled {
  background: #888;
  cursor: not-allowed;
}
