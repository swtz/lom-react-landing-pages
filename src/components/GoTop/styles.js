import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    display: flex;
    align-items: center;
    justify-content: center;
    bottom: 2rem;
    right: 2rem;
    width: 4rem;
    height: 4rem;
    z-index: 6;
  `}
`;
