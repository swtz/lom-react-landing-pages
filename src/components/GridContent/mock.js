export default {
  title: 'Any title as you wish',
  html: `<h1>Non voluptas eligendi 33 beatae sint est quod exercitationem! </h1><p>Lorem ipsum dolor sit amet. Et asperiores explicaboEst earum ea accusamus similique aut ipsa totam aut laborum quidem rem vero quisquam est velit labore? Eos veniam atqueUt consequatur id eius enim. Sit Quis adipisci <em>Eum eaque et nemo praesentium 33 voluptas aperiam et atque explicabo</em> eos vitae eius At saepe aspernatur. Ut quaerat cumque aut suscipit dolorumad iste et nisi autem et nobis galisum! </p><h2>Ut omnis maiores qui aperiam omnis et molestias ipsa. </h2><p>Aut esse sunt <em>In aliquam et facere iure</em>. Ut minima oditEt assumenda qui nesciunt rerum est deserunt aliquam non laborum exercitationem et minus dolorem! Qui rerum quisquam sit velit quibusdamEt iusto eos pariatur doloribus et temporibus saepe in dolorem beatae. Hic quia suntIn tempora non voluptatem debitis rem labore nihil est cumque sunt quo atque voluptas. </p><h3>Ab suscipit atque aut animi quod. </h3><p>Aut labore magnam <em>33 voluptatibus aut iste iste et accusamus iste</em>. Vel accusantium commodiEst sequi aut provident quia et nihil quae sed fuga aspernatur in praesentium voluptas ut incidunt laborum. Non porro quia ut architecto illum <strong>Ut laborum At dignissimos voluptas qui quae nulla rem laboriosam doloribus</strong> qui voluptatem accusamus! </p>
`,
  background: false,
};
