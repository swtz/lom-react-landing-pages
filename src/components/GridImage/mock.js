export default {
  background: false,
  title: 'My grid',
  description:
    'Lorem ipsum dolor sit amet. Et asperiores explicaboEst earum ea accusamus similique aut ipsa totam aut laborum quidem rem vero quisquam est velit labore? Eos veniam atqueUt consequatur id eius enim.',
  grid: [
    {
      altText: 'Teste 1',
      srcImg: '/assets/images/image_001.webp',
    },
    {
      altText: 'Teste 2',
      srcImg: '/assets/images/image_002.webp',
    },
    {
      altText: 'Teste 3',
      srcImg: '/assets/images/image_003.webp',
    },
    {
      altText: 'Teste 4',
      srcImg: '/assets/images/image_004.webp',
    },
    {
      altText: 'Teste 5',
      srcImg: '/assets/images/image_005.webp',
    },
    {
      altText: 'Teste 6',
      srcImg: '/assets/images/image_006.webp',
    },
  ],
};
