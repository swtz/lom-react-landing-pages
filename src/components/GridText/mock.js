export default {
  background: false,
  title: 'My grid',
  description:
    'Lorem ipsum dolor sit amet. Et asperiores explicaboEst earum ea accusamus similique aut ipsa totam aut laborum quidem rem vero quisquam est velit labore? Eos veniam atqueUt consequatur id eius enim.',
  grid: [
    {
      title: 'Teste 1',
      description:
        'Lorem ipsum dolor sit amet. Et asperiores explicaboEst earum ea accusamus similique aut ipsa totam aut laborum quidem rem vero quisquam est velit labore? Eos veniam atqueUt consequatur id eius enim.',
    },
    {
      title: 'Teste 2',
      description:
        'Lorem ipsum dolor sit amet. Et asperiores explicaboEst earum ea accusamus similique aut ipsa totam aut laborum quidem rem vero quisquam est velit labore? Eos veniam atqueUt consequatur id eius enim.',
    },
    {
      title: 'Teste 3',
      description:
        'Lorem ipsum dolor sit amet. Et asperiores explicaboEst earum ea accusamus similique aut ipsa totam aut laborum quidem rem vero quisquam est velit labore? Eos veniam atqueUt consequatur id eius enim.',
    },
  ],
};
