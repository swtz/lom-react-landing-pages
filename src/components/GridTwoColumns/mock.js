export default {
  title: 'Grid Two Columns',
  text: `Lorem ipsum dolor sit amet consectetur adipisicing elit. Atque libero fugiat nobis corporis quos enim ea molestias modi et, exercitationem ab a praesentium est numquam recusandae sit, omnis similique asperiores.`,
  srcImg: 'assets/images/javascript.svg',
};
