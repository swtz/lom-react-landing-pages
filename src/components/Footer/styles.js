import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';
import { Container as SectionContainer } from '../SectionContainer/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    text-align: center;
    border-top: 0.1rem solid ${theme.colors.mediumGray};

    a {
      color: inherit;
      text-decoration: none;
    }
    
    & ${TextComponent} {
      font-size: ${theme.font.sizes.small};
    }

    & ${SectionContainer} {
      padding-top: ${theme.font.sizes.small};
      padding-bottom: ${theme.font.sizes.small};
    }
  `}
`;
