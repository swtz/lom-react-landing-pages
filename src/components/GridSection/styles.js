import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css``}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
  counter-reset: grid-counter;
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
  ${HeadingContainer} {
    position: relative;
    left: 5rem;
  }

  ${HeadingContainer}::before {
    counter-increment: grid-counter;
    content: counter(grid-counter);
    position: absolute;
    font-size: 7rem;
    top: -3rem;
    left: -5rem;
    transform: rotate(15deg);
  }
  `}
`;
