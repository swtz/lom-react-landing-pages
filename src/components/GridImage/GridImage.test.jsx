import { renderTheme } from '../../styles/render-theme';
import { GridImage } from '.';
import mock from './mock';

describe('<GridText />', () => {
  it('should render grid component', () => {
    const { container } = renderTheme(<GridImage {...mock} />);
    expect(container).toMatchSnapshot();
  });

  it('should render grid component without background', () => {
    const { container } = renderTheme(
      <GridImage {...mock} background={undefined} />,
    );
    expect(container).toMatchSnapshot();
  });
});
