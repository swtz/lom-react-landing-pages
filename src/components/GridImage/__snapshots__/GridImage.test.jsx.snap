// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridText /> should render grid component 1`] = `
.c3 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c5 {
  font-size: 2.4rem;
}

.c2 > .c4 {
  margin-bottom: 6.4rem;
}

.c6 {
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(280px,1fr));
  gap: 3.2rem;
}

.c7 {
  overflow: hidden;
}

.c8 {
  width: 100%;
}

.c0 {
  background: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

@media (max-width:768px) {
  .c3 {
    font-size: 4.0rem;
  }
}

@media (min-width:1024px) {
  .c8 {
    -webkit-transition: all 300ms ease-in-out;
    transition: all 300ms ease-in-out;
  }

  .c8:hover {
    -webkit-transform: scale(1.2) rotate(10deg);
    -ms-transform: scale(1.2) rotate(10deg);
    transform: scale(1.2) rotate(10deg);
  }
}

@media (max-width:768px) {
  .c1 {
    padding: 1.6rem;
  }
}

@media (max-width:414px) {
  .c1 {
    padding: .4rem;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h2
          class="c3"
        >
          My grid
        </h2>
        <p
          class="c4 c5"
        >
          Lorem ipsum dolor sit amet. Et asperiores explicaboEst earum ea accusamus similique aut ipsa totam aut laborum quidem rem vero quisquam est velit labore? Eos veniam atqueUt consequatur id eius enim.
        </p>
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            <img
              alt="Teste 1"
              class="c8"
              src="/assets/images/image_001.webp"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Teste 2"
              class="c8"
              src="/assets/images/image_002.webp"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Teste 3"
              class="c8"
              src="/assets/images/image_003.webp"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Teste 4"
              class="c8"
              src="/assets/images/image_004.webp"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Teste 5"
              class="c8"
              src="/assets/images/image_005.webp"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Teste 6"
              class="c8"
              src="/assets/images/image_006.webp"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<GridText /> should render grid component without background 1`] = `
.c3 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c5 {
  font-size: 2.4rem;
}

.c2 > .c4 {
  margin-bottom: 6.4rem;
}

.c6 {
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(280px,1fr));
  gap: 3.2rem;
}

.c7 {
  overflow: hidden;
}

.c8 {
  width: 100%;
}

.c0 {
  background: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

@media (max-width:768px) {
  .c3 {
    font-size: 4.0rem;
  }
}

@media (min-width:1024px) {
  .c8 {
    -webkit-transition: all 300ms ease-in-out;
    transition: all 300ms ease-in-out;
  }

  .c8:hover {
    -webkit-transform: scale(1.2) rotate(10deg);
    -ms-transform: scale(1.2) rotate(10deg);
    transform: scale(1.2) rotate(10deg);
  }
}

@media (max-width:768px) {
  .c1 {
    padding: 1.6rem;
  }
}

@media (max-width:414px) {
  .c1 {
    padding: .4rem;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h2
          class="c3"
        >
          My grid
        </h2>
        <p
          class="c4 c5"
        >
          Lorem ipsum dolor sit amet. Et asperiores explicaboEst earum ea accusamus similique aut ipsa totam aut laborum quidem rem vero quisquam est velit labore? Eos veniam atqueUt consequatur id eius enim.
        </p>
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            <img
              alt="Teste 1"
              class="c8"
              src="/assets/images/image_001.webp"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Teste 2"
              class="c8"
              src="/assets/images/image_002.webp"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Teste 3"
              class="c8"
              src="/assets/images/image_003.webp"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Teste 4"
              class="c8"
              src="/assets/images/image_004.webp"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Teste 5"
              class="c8"
              src="/assets/images/image_005.webp"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Teste 6"
              class="c8"
              src="/assets/images/image_006.webp"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
